examples/measure/rs-measure.cpp

```diff
@@ -0,0 +1,553 @@
+                std::map<pixel, pixel> parent;
+                // Priority queue holds pixels (ordered by their distance)
+                std::priority_queue<dv, std::vector<dv>, std::greater<dv>> q;
+
+                // Initialize the source pixel:
+                dist[src] = 0.f;
+                parent[src] = token;
+                q.emplace(0.f, src);
+
+                // To save calculation we apply a heuristic
+                // Don't visit pixels that are too far away in 2D space
+                // It is very rare for objects in 3D space to violate this
+                auto max_2d_dist = dist_2d(src, trg) * 1.2;
+
+                while (!q.empty() && alive)
+                {
+                    // Fetch the closest pixel from the queue
+                    pixel u = q.top().second; q.pop();
+
+                    // If we reached the max radius, don't continue to expand
+                    if (dist_2d(src, u) > max_2d_dist) continue;
+
+                    // Fetch the list of neighboring pixels
+                    auto n = neighbors(depth, u);
+                    for (auto&& v : n)
+                    {
+                        // If this pixel was not yet visited, initialize
+                        // its distance as +INF
+                        if (dist.find(v) == dist.end()) dist[v] = INFINITY;
+
+                        // Calculate distance in 3D between the two neighboring pixels
+                        auto d = dist_3d(intrinsics, depth, u, v);
+                        // Calculate total distance from source
+                        auto total_dist = dist[u] + d;
+
+                        // If we encounter a potential improvement,
+                        if (dist[v] > total_dist)
+                        {
+                            // Update parent and distance
+                            parent[v] = u;
+                            dist[v] = total_dist;
+                            // And re-visit that pixel by re-introducing it to the queue
+                            q.emplace(total_dist, v);
+                        }
+                    }
+                }
+
+                {
+                    // Write the shortest-path to the path variable
+                    std::lock_guard<std::mutex> lock(path_mutex);
+                    total_dist = dist[trg];
+                    path.clear();
+                    // Iterate until encounter token special pixel
+                    while (trg != token)
+                    {
+                        // Handle the case we didn't find a path
+                        if (trg == parent[trg]) break;
+
+                        path.emplace_back(trg);
+                        trg = parent[trg];
+                    }
+                }
+            }
+        }
+    });
+
+    while(app) // Application still alive?
+    {
+        // Fetch the latest available post-processed frameset
+        static rs2::frameset current_frameset;
```

Comment:
Why use `static` and not automatic variable (outside the loop)?
Actually, next line is `poll_for_frames` so why is this `static` and not simply scoped in the loop? Am I missing something?